### ISSUE TITLE
Can’t install TensorRT 8.6.1.6 from network repo

### ISSUE THREAD

**Issue by Hato1** [NVIDIA/TensorRT#3774]
## Description

Installing TensorRT 8.6.1.6 via the network repo no longer works. It used to as of sometime in March.

I found this related issue workaround that would probably work here: https://github.com/NVIDIA/TensorRT/issues/1741#issuecomment-1954319329

## Steps To Reproduce

I can reproduce this both on my system and several of the official CUDA containers. Here are the simplest repro steps:

```bash
docker run --interactive nvcr.io/nvidia/cuda:12.0.1-base-ubuntu20.04
apt-get update
apt-get install tensorrt=8.6.1.6-1+cuda12.0
```

Output:
```
Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 tensorrt : Depends: libnvinfer-bin (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
            Depends: libnvinfer-dev (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
            Depends: libnvinfer-lean-dev (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
            Depends: libnvinfer-plugin-dev (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
            Depends: libnvinfer-vc-plugin-dev (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
            Depends: libnvinfer-dispatch-dev (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
            Depends: libnvparsers-dev (= 8.6.1.6-1+cuda12.0) but it is not going to be installed
            Depends: libnvonnxparsers-dev (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
            Depends: libnvinfer-samples (= 8.6.1.6-1+cuda12.0) but 10.0.0.6-1+cuda12.4 is to be installed
E: Unable to correct problems, you have held broken packages.
```

**Comment by lix19937** [comment]
NOTE:    
msg `The following packages have unmet dependencies`     
see available versions 
```
sudo apt-cache madison tensorrt  
```

**Comment by zerollzeng** [comment]
Please refer to https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html#overview

**Comment by Hato1** [comment]
I'm following the [latest stable version documentation](https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-861/install-guide/index.html#maclearn-net-repo-install).

The [latest early access (v10) release documentation](https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html#maclearn-net-repo-install) only states how to install v10. I notice a new big list of apt installs if you aren't using the latest supported CUDA.

Despite using the latest supported CUDA for stable TRT, running the big list of apt installs (omitting the packages suffixed with '10') with `version="8.6.1.6-1+cuda12.0"` does operate as a workaround.

Is this breaking change intended behavior? Can we rely on this new install method to work in the future?

**Comment by lix19937** [comment]
The best way to install tensorrt(py, cpp) , you load `nv-tensorrt-repo-ubuntu2004-cudaXX.Y-trtX.Y.Z.W-x86-host-ga-yyyymmdd.deb`, and then dpkg -i install.

**Comment by zeroepoch** [comment]
We recently corrected the documentation to list all packages that were available for TRT 8.6.1 in the 8.6 install guide. See https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-861/install-guide/index.html#maclearn-net-repo-install. The other option is as @lix19937 said, which is to use a local repo instead of the network repo. This issue is due to a limitation with `apt`. Apt's resolver is not smart enough to realize older versions of packages, which are available, could be used to satisfy dependencies.

`dnf` (on RedHat) is a little smarter in this regard in that it will match the versions of the packages. Although, it also tries to install the latest CUDA version for the older release unless every package version is specified.
